DM-28724: Start a page for IDF (#467)
lsst-dm/dm_dev_guide: idf/overview.rst

##############################
Interim Data Facility Overview
##############################

> Pre-operations activities may be carried out on Interim Data Facility (IDF) hosted on `Google Cloud Platform <https://cloud.google.com/>`_.

Reviewer: ```suggestion
Pre-operations activities may be carried out on the Interim Data Facility (IDF) hosted on the `Google Cloud Platform <https://cloud.google.com/>`_.
```

The `Google Cloud documentation site <https://cloud.google.com/docs/overview>`_ provides an overview of its products and services.

> IDF is considered a production environment to support Data Preview releases.
> Development and Commissioning work continue on `LDF <../services/lsst-devl.rst>`_.

Reviewer: ```suggestion
The IDF is considered a production environment to support Data Preview releases.
Development and Commissioning work continue on the Construction-funded `LSST Data Facility (LDF) <../services/lsst-devl.rst>`_.
```

Login credentials
=================

> Rubin staff working on deploying and operating Data Production services receives a ``username@lsst.cloud`` identity.

Reviewer: ```suggestion
Rubin staff working on deploying and operating Data Production services receive a ``username@lsst.cloud`` identity.
```

This IDF cloud identity is not an email and is independent from other Rubin accounts.

> Not everybody in the Data Production department needs this login; contact your pre-operations team leads if in doubt.

Reviewer: ```suggestion
Not everybody in the Data Production department needs this login; contact your pre-operations team lead(s) if in doubt.
```

> 2FA is required for the IDF identity.

Reviewer: ```suggestion
Two-factor authentication (2FA) is required for the IDF identity.
```

See :ref:`idf-internal-support` for lost password or login issues.

> Environments

Reviewer: Why "Environments" and not "Projects"?

Author: I wanted something that's broader than the GCP "projects", as part of this session is about using Terraform in idf_deploy. But maybe this session can be split into two sessions?

============

> On GCP, resources are organized into a base-level entity called "project".

Reviewer: ```suggestion
On GCP, resources are organized into a base-level entity called a "project".
```

A project is linked to billing, permissions, and other settings.
Projects are grouped in folders.
We have six top-level folders:

- ``Science Platform``
- ``QServ``
- ``Processing``
- ``SQuaRE``
- ``Shared Services``
- ``Scratch``

> For each top-level folders except ``Shared Services`` and ``Scratch``, we have three sub-folders: ``Dev``, ``Integration``, and ``Production``.

Reviewer: ```suggestion
For each top-level folder except ``Shared Services`` and ``Scratch``, we have three sub-folders: ``Dev``, ``Integration``, and ``Production``.
```

``Production`` is intended to host services used by end users.
``Dev`` and ``Integration`` provide testing and staging environments where one can deploy near-production services.

> All environments on IDF are managed with `Terraform <https://www.terraform.io/>`_ infrastructure-as-code tool in the `idf_deploy <https://github.com/lsst/idf_deploy>`_ repo.
> See the documentations in `idf_deploy <https://github.com/lsst/idf_deploy>`_'s `run book <https://github.com/lsst/idf_deploy/tree/master/runbook>`_.

Reviewer: ```suggestion
All environments on IDF are managed with `Terraform <https://www.terraform.io/>`_, an infrastructure-as-code tool, via configuration files in the `idf_deploy <https://github.com/lsst/idf_deploy>`_ repo.
See the documentation in idf_deploy's `run book <https://github.com/lsst/idf_deploy/tree/master/runbook>`_.
```

> GitHub Actions are used to automatically deploy GCP resources on IDF.
> All resources should be deployed via `idf_deploy <https://github.com/lsst/idf_deploy>`_ or layers on top of it, so operations can be repeated consistently.

Reviewer: ```suggestion
GitHub Actions are used to automatically deploy GCP resources on IDF in response to pull requests on the configuration files.
All resources should be deployed this way so that operations can be repeated consistently.
```

The only exception to using `idf_deploy <https://github.com/lsst/idf_deploy>`_ is the ``Scratch`` folder.

> Projects inside the ``Scratch`` folder are for short lived testing or cloud training; they may be removed with a short notice.

Reviewer: ```suggestion
Projects inside the ``Scratch`` folder are for short-lived testing or cloud training; they may be removed on short notice.
No user-facing services should live in the ``Scratch`` folder.
```

> Contact your pre-operations team leads if your work requires a scratch project.

Reviewer: ```suggestion
Contact your pre-operations team lead(s) if your work requires a scratch project.
```

> By default, the ``us-central1`` region should be used.

Reviewer: ```suggestion
All IDF resources should live in the ``us-central1`` region.
```

Permissions
===========

Permissions are managed via Google Groups configured in the `idf_deploy <https://github.com/lsst/idf_deploy>`_ repo.
Group memberships are managed manually by the administrators.

> See :ref:`idf-internal-support` for requesting to be added to an existing group.

Reviewer: ```suggestion
See :ref:`idf-internal-support` to request to be added to an existing group.
```

Monitoring
==========

> A number of dashboards have been configured and they are viewable in Cloud Monitoring; see `Monitoring and Logging in the run book <https://github.com/lsst/idf_deploy/blob/master/runbook/monitoring-logging.md>`_.

Reviewer: ```suggestion
A number of dashboards have been configured in Cloud Monitoring; for more information, see `Monitoring and Logging in the run book <https://github.com/lsst/idf_deploy/blob/master/runbook/monitoring-logging.md>`_.
```

.. _idf-internal-support:

Support (interim)
=================

> Questions can be sent to the #ops-google-idf Slack channel.

Reviewer: ```suggestion
Questions can be sent to the `#ops-google-idf <https://lsstc.slack.com/archives/CC82DT23Y>`_ Slack channel.
```

> For requests or infrastructure service issues, file a `JIRA <https://jira.lsstcorp.org/>`_ ticket, add the ``idf`` label, and mention the ticket in the #ops-google-idf Slack channel.

Reviewer: ```suggestion
For requests or infrastructure service issues, file a `Jira <https://jira.lsstcorp.org/>`_ ticket, add the ``idf`` label, and mention the ticket in the #ops-google-idf Slack channel.
```
Should the ticket be in DM, IHS, or some other Jira project?

Author: I feel we might be just trying out this ticketing model so maybe we shouldn't create a new Jira project yet.  Maybe IHS is more appropriate but getting IDF tickets mixed up with other LDF/IT tickets doesn't sound fun to me. DM sounds the easiest way to start if we don't have to worry about the color-of-the-money issue?

Opinions?

Reviewer: I'm fine with DM; the main thing is picking something and documenting which it is.

Author: I'm going with DM for now then.   Added to the doc.
